### PR TITLE
Remove unsynchronised run loop

### PR DESCRIPTION
**Pull request by mateo-mro**
# Description
Since now all the e2e tests are working in sync-to-blockchain mode (see https://github.com/cowprotocol/services/pull/3040) , we can remove the unsynchronized mode.

# Changes
- Remove unsynchronized mode

## How to test
1. Regression tests

### CODE REVIEW

File: crates/autopilot/src/maintenance.rs
Comment (MartinquaXD) [inline crates/autopilot/src/maintenance.rs:125]: The whole background task stuff can be removed now.

File: crates/autopilot/src/shadow.rs
Comment (MartinquaXD) [inline crates/autopilot/src/shadow.rs:77]: Small comment explaining that we use this as a synchronization mechanism would be nice.

**Review by sunce86 — APPROVED** [review]
LG

**Review by squadgazzz — APPROVED** [review]
🔪

**Review by MartinquaXD — APPROVED** [review]
With the background task `--auction-update-interval` is now also no longer in use. Other than that the PR looks good.
Approved assuming that will also get removed before merging.

**Pull request merged**
